core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -320,6 +333,56 @@ protected void validateNoNewDeletesForDataFiles(TableMetadata base, Long startin
     }
   }
 
+  /**
+   * Validates that no delete files matching a filter have been added to the table since a starting snapshot.
+   *
+   * @param base table metadata to validate
+   * @param startingSnapshotId id of the snapshot current at the start of the operation
+   * @param dataFilter an expression used to find new conflicting delete files
+   * @param caseSensitive whether expression evaluation should be case-sensitive
+   */
+  protected void validateNoNewDeleteFiles(TableMetadata base, Long startingSnapshotId,
+                                          Expression dataFilter, boolean caseSensitive) {
+    // if there is no current table state, no files have been added
+    if (base.currentSnapshot() == null) {
```

Comment:
Can you add the check for `base.formatVersion() < 2`?

Reply:
Added to both methods. Could you check, @rdblue?